configuration/postgres/postgres.go

```diff
@@ -0,0 +1,417 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"k8s.io/utils/strings/slices"
+
+	"github.com/dapr/components-contrib/configuration"
+	"github.com/dapr/kit/logger"
+)
+
+type ConfigurationStore struct {
+	metadata             metadata
+	client               *pgxpool.Pool
+	logger               logger.Logger
+	configLock           sync.Mutex
+	subscribeStopChanMap map[string]interface{}
+	ActiveSubscriptions  map[string]*subscription
+}
+
+type subscription struct {
+	uuid string
+	keys []string
+}
+
+const (
+	configtablekey               = "table"
+	connMaxIdleTimeKey           = "connMaxIdleTime"
+	connectionStringKey          = "connectionString"
+	pgNotifyChannelKey           = "pgNotifyChannel"
+	listenTemplate               = "listen %s"
+	unlistenTemplate             = "unlisten %s"
+	payloadDataKey               = "data"
+	ErrorMissingTableName        = "missing postgreSQL configuration table name"
+	ErrorMissingTable            = "postgreSQL configuration table - '%s' does not exist"
+	InfoStartInit                = "initializing postgreSQL configuration store"
+	ErrorMissingConnectionString = "missing postgreSQL connection string"
+	ErrorAlreadyInitialized      = "postgreSQL configuration store already initialized"
+	ErrorMissingMaxTimeout       = "missing PostgreSQL maxTimeout setting in configuration"
+	QueryTableExists             = "SELECT EXISTS (SELECT FROM pg_tables where tablename = $1)"
+	ErrorTooLongFieldLength      = "field name is too long"
+	maxIdentifierLength          = 64 // https://www.postgresql.org/docs/current/limits.html
+)
+
+var (
+	allowedChars           = regexp.MustCompile(`^[a-zA-Z0-9./_]*$`)
+	defaultMaxConnIdleTime = time.Minute * 30
+)
+
+func NewPostgresConfigurationStore(logger logger.Logger) configuration.Store {
+	logger.Debug("Instantiating PostgreSQL configuration store")
+	return &ConfigurationStore{
+		logger:               logger,
+		subscribeStopChanMap: make(map[string]interface{}),
+		configLock:           sync.Mutex{},
```

Comment:
Nit: Can be removed

Reply:
Done - 
https://github.com/dapr/components-contrib/blob/86b77cc269f728e552f2812ef0ba1a0d3607095d/configuration/postgres/postgres.go#L76-L81